extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -578,7 +504,7 @@ public StreamType getStreamType() throws ParsingException {
     @Override
     public StreamInfoItem getNextStream() throws ExtractionException {
         assertPageFetched();
-        if (isAgeRestricted) {
+        if (getAgeLimit() == 18) {
```

Comment:
```suggestion
        if (getAgeLimit() != NO_AGE_LIMIT) {
```

Reply:
That's better indeed, though `getAgeLimit()` will only ever return `18`.